website/src/_posts/2019-04-19-beam-2.12.0.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title:  "Apache Beam 2.12.0"
+date:   2019-04-19 00:00:01 -0800
+excerpt_separator: <!--more-->
+categories: blog
+authors:
+        - apilloud
+
+---
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+We are happy to present the new 2.12.0 release of Beam. This release includes both improvements and new functionality.
+See the [download page]({{ site.baseurl }}/get-started/downloads/#2120-2019-04-19) for this release.<!--more-->
+For more information on changes in 2.12.0, check out the
+[detailed release notes](https://jira.apache.org/jira/secure/ReleaseNote.jspa?projectId=12319527&version=12344944).
+
+## Highlights
+
+### I/Os
+
+* Implement a BigQuery custom sink for Python SDK.
+* Add support to specify a query in CassandraIO for Java SDK.
+
+### New Features / Improvements
+
+* Enable Bundle Finalization in Python SDK.
+* Add support to the Java SDK harness to merge windows.
```

Comment:
I think we should mention `GenerateSequence` as first cross-language transform: https://issues.apache.org/jira/browse/BEAM-6730 - it may also be good to clarify what we have as of 2.12 in the ticket @mxm 

@aaltay is there anything that should be mentioned here regarding Python3 ?

Reply:
Agree, we could mention cross-language transforms as a new feature.  

For [Flink](https://jira.apache.org/jira/issues/?jql=project%20%3D%20BEAM%20AND%20fixVersion%20%3D%202.12.0%20AND%20component%20%3D%20runner-flink), I'd also mention Kafka Sink EOS support.

Reply:
Python 3, things I am aware of
- bug fixes related to serialization. No need to mention this separately.
- py 3.6 and 3.7 workloads enabled -- still more experimental than 3.5, worth mentioning.
- BQ and Avro, was there a default change related to fastavro since last release?
- Does datastore io change happened?

@tvalentyn for the questions.

Reply:
> Agree, we could mention cross-language transforms as a new feature.
> 

@mxm I would suggest we clarify what exactly the user can expect to work in 2.12 here and also on the JIRA ticket: https://issues.apache.org/jira/browse/BEAM-6730

Reply:
BQ, Avro, Datastore changes didn't make it into 2.12.
The 3.6, 3.7 change is specific to Dataflow runner only. We could say:
"Beam 2.12 supports starting Dataflow pipelines under Python 3.6, 3.7, however 3.5 remains the only recommended minor version for Dataflow runner. In addition to [announced 2.11 limitations](https://issues.apache.org/jira/browse/BEAM-1251?focusedCommentId=16789854&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16789854), Beam typehint annotations are currently not supported on Python >= 3.6."

Reply:
@tweise I've added a summary in the JIRA issue. It is too long for the release notes but perhaps we could include something like this:

```
- Adds experimental support for cross-language transforms, 
  please see [BEAM-6730](https://issues.apache.org/jira/browse/BEAM-6730) 
```

```
- Adds support in the Flink Runner for exactly-once Writes with KafkaIO
```